packages/x-date-pickers/src/AdapterDateFnsJalali/index.ts

```diff
@@ -23,6 +23,30 @@ const formatTokenMap: FieldFormatTokenMap = {
   // Day of the month
   d: 'day',
   dd: 'day',
+  do: 'day',
```

Comment:
Fun fact, the `1st` / `2nd` format has it's letters before the digit here (it's a prefix) and it still works fine :ok_hand: